index/scorch/snapshot_index.go

```diff
@@ -471,16 +471,35 @@ func (is *IndexSnapshot) Document(id string) (rv index.Document, err error) {
 	return rvd, nil
 }
 
+func (is *IndexSnapshot) localDocNumFromGlobal(segmentIndex int, docNum uint64) uint64 {
```

Comment:
Can you add some commentary around what this does and for the pre-existing API `segmentIndexAndLocalDocNumFromGlobal` as well please.